editor/components/block-list/block.js

```diff
@@ -348,6 +354,96 @@ export class BlockListBlock extends Component {
 		this.setState( { error } );
 	}
 
+	onDragStart( event ) {
+		const dragInset = document.getElementById( `block-drag-inset-${ this.props.uid }` );
+		const block = document.getElementById( `block-${ this.props.uid }` );
+
+		// Closure to remove the cloned node from the DOM (fired within timeout below)
+		const removeBlockClone = ( _blockList, _cloneWrapper ) => {
+			return () => {
+				_blockList.removeChild( _cloneWrapper );
+			};
+		};
+
+		// Closure to hide the visible block and show inset in its place (fired within timeout below)
+		const showInset = () => {
+			return () => {
+				this.setState( state => Object.assign( {}, state, { dragging: true } ) );
+				document.body.classList.add( 'dragging' );
+			};
+		};
+
+		event.dataTransfer.setData(
+			'text',
+			JSON.stringify( {
+				uid: this.props.uid,
+				fromIndex: this.props.order,
+				type: BLOCK_REORDER,
+			} )
+		);
+
+		if ( 'function' === typeof event.dataTransfer.setDragImage ) {
+			const blockList = block.parentNode;
+			const cloneWrapper = document.createElement( 'div' );
+			const clone = block.cloneNode( true );
+			const blockRect = block.getBoundingClientRect();
+
+			// 1. Clone the current block, style it, and spawn over original block.
+
+			clone.id = `clone-${ block.id }`;
+
+			cloneWrapper.classList.add( 'editor-block-list__block-clone' );
+			// 40px padding for the shadow.
+			cloneWrapper.style.width = `${ blockRect.width + 40 }px`;
```

Comment:
Do we need all those numbers (shadow paddings, top/lefts below) to be magic? Is there a better place for them?

Reply:
Agreed. I will try and find a better place for them if this persists in the final version.